stellarphot/core.py

```diff
@@ -62,3 +70,305 @@ def __init__(self, gain=1.0 * u.electron / u.adu,
         self.gain = gain
         self.read_noise = read_noise
         self.dark_current = dark_current
+
+
+@autocheck_required_columns
+class BaseEnhancedTable(BaseTimeSeries):
+    """
+    A class to validate an `astropy.table.Table` table of astronomical data during
+    creation.
+
+    This is based on the `astropy.timeseries.BaseTimeSeries` class to
+    allow inheriting its ability to have required columns (but not requiring
+    the time column).  We extend this to allow for checking of the units for
+    the columns to match the table_description.
+
+    Parameters
+    ----------
+
+    table_description: dict
+        This is a dictionary where each key is a required table column name
+        and the value corresponding to each key is the required dtype
+        (can be None).  This is used to check the format of the input data
+        table.
+
+    data: `astropy.table.Table`
+        A table containing astronomical data of interest.  This table
+        will be checked to make sure all columns listed in table_description
+        exist and have the right runits. However, additional columns that
+        may be present in data but are not listed in table_description will
+        NOT be removed.
+
+    Attributes
+    ----------
+    data: `astropy.table.Table`
+        A table formatted to match the table_description formatting information
+        but containing no data.
+    """
+
+    def __init__(self, table_description, data):
+        # Handle parameters
+        self._table_description = table_description
```

Comment:
See below for a suggestion about this. I think the `TypeError` check should be a little different and be up here before the rest of the processing happens.

Also, I think as currently written `self._table_description` will modify the `dict` that the user passes in. Haven't checked that, though.

Reply:
I fixed this by copying the dict.